### PR TITLE
Emagged defib and electric chairs no longer harms slime people

### LINKED ISSUE #23975: Emagged defib harms slime people
(WEB REPORT BY: awadwduwu REMOTE: 206.221.180.138:7777)
# Revision
8b0eec6c7784ed74f405ced4dc443c6bcdb6daf3

# Description
Slime people, who are immune to shocks, still get shocked and hurt by this
# Steps to Reproduce
Get shocked as slime person
# What you Expected
It heals
# What Actually Happened
Straight to orange

### PR DESCRIPTION
**Pull request by Zth--**
Closes #23975 

~I'm forgetting something. Gonna take a bath while I try to remember~
done

:cl:
 * bugfix: Emagged defibs and electric chairs no longer electrocutes slime people

### CODE REVIEW

File: code/game/objects/items/weapons/defib.dm [hunk 1]
```diff
@@ -105,16 +105,13 @@
...
+	var/damage = rand(30, 60)
+	if (!target.electrocute_act(damage, src, def_zone = LIMB_CHEST))
+		return
 	var/datum/organ/internal/heart/heart = target.get_heart()
 	if(heart)
 		heart.damage += rand(5,60)
-	target.visible_message("<span class='danger'>[target] has been shocked in the chest with the [src] by [user]!</span>")
-	var/incapacitation_duration = rand(6, 12)
-	target.Knockdown(incapacitation_duration)
-	target.Stun(incapacitation_duration)
```
Comment (Probe1) [inline code/game/objects/items/weapons/defib.dm:114]: Why was all this discarded?

Comment (Zth--) [reply code/game/objects/items/weapons/defib.dm:114]: It's now being handled by electrocute_act() (that's where the slime flag check lives)

**Comment by tayachiya** [comment]
Maybe that the electric chair also burns slime people? I know it's a very snowflakey object, but still

**Comment by Zth--** [comment]
Yes, I will make it so electrocute_act() takes a `incapacitation_duration` parameter in time as seconds to deal with hardcoded stuns so I can fix the electric chair properly.
I'm thinking of some other complex solutions but that will cut it.

Gonna do it tonight probably, so it's WIP until then

**Comment by Zth--** [comment]
Done. Electric chairs are still very snowflakey but at least they call `electrocute_act` now

**Pull request merged**